src/reclaim_python_sdk/witness.py

```
@@ -4,20 +4,31 @@
     Returns:
         str: Hex string identifier
     """
-    string = f"{info.provider}\n{info.parameters}\n{info.context}"
+    canonical_context = info.context or ''
+    if canonical_context:
+        try:
+            ctx = json.loads(canonical_context)
+            canonical_bytes = canonicalize(ctx)
+            canonical_context = canonical_bytes.decode('utf-8') if isinstance(canonical_bytes, bytes) else canonical_bytes
+        except (json.JSONDecodeError, TypeError):
+            raise ValueError('unable to parse non-empty context. Must be JSON')
```

Ruff (0.15.10)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)